tests/rag_answers/custom_deepeval/metrics/coherence/test_coherence.py

```diff
@@ -0,0 +1,184 @@
+        ) as mock_logs:
+            mock_logs.return_value = "verbose"
+            await metric.a_measure(test_case)
+
+            expected_steps = [
+                "Rubric Score: 3",
+                f"Score: {metric.score}",
+                "Reason: Clear enough",
+            ]
+            mock_logs.assert_called_once_with(metric, steps=expected_steps)
+            assert metric.verbose_logs == "verbose"
+
+    @pytest.mark.asyncio
+    @patch(
+        "govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.coherence.coherence.capture_metric_type"
+    )
+    async def test_success_threshold_and_reason_toggle(
+        self,
+        mock_capture,
+        mock_native_model: Mock,
+        test_case: LLMTestCase,
+    ):
+        mock_native_model.a_generate = AsyncMock(
+            return_value=(CoherenceJudgement(score=5, reason="Great"), 0.0)
+        )
+        metric = CoherenceMetric(
+            model=mock_native_model,
+            threshold=0.75,
+            include_reason=False,
+        )
+
+        score = await metric.a_measure(test_case)
+
+        assert score == 1.0
+        assert metric.success is True
+        assert metric.reason is None
+        mock_capture.assert_called_once_with(
+            metric.__name__, async_mode=True, in_component=False
+        )
+
+    @pytest.mark.asyncio
+    async def test_typeerror_triggers_json_fallback(
+        self,
+        mock_non_native_model: Mock,
+        test_case: LLMTestCase,
+    ):
+        json_response = '{"score": 2, "reason": "Fallback path"}'
+        mock_non_native_model.a_generate = AsyncMock(
+            side_effect=[TypeError("no schema"), json_response]
+        )
+        metric = CoherenceMetric(model=mock_non_native_model)
+
+        score = await metric.a_measure(test_case)
+
+        assert score == pytest.approx(0.25)
+        assert metric.rubric_score == 2
+        assert metric.reason == "Fallback path"
+        assert metric.evaluation_cost is None  # non-native models skip cost tracking
+
+        assert mock_non_native_model.a_generate.call_count == 2
+        first_call = mock_non_native_model.a_generate.call_args_list[0]
+        second_call = mock_non_native_model.a_generate.call_args_list[1]
+        assert "schema" in first_call.kwargs
+        assert "schema" not in second_call.kwargs
+
+    @pytest.mark.asyncio
+    async def test_score_is_clamped_into_range(
+        self,
+        mock_non_native_model: Mock,
+        test_case: LLMTestCase,
+    ):
+        mock_non_native_model.a_generate = AsyncMock(
+            return_value=CoherenceJudgement(score=999, reason="Out of bounds")
+        )
+        metric = CoherenceMetric(model=mock_non_native_model)
+
+        score = await metric.a_measure(test_case)
+
+        assert score == 1.0
+        assert metric.rubric_score == 5
+
+    @pytest.mark.asyncio
+    async def test_error_sets_nan_score(
```

Comment:
I don't think we need this test as it's now just asserting an exception is unhandled. I assume it's left behind from work on error handling